addons/project/tests/test_project_subtasks.py

```diff
@@ -354,3 +354,29 @@ def test_get_all_subtasks(self):
             self.assertEqual(parent_id, task.id, "The key should be the common ancestor")
             self.assertEqual(set(subtask_ids), set(all_subtasks.ids),
                              "All subtasks linked to the common ancestor should be returned by _get_subtask_ids_per_task_id method")
+
+    def test_subtask_is_visible_after_archiving(self):
+        """
+            Check if `display_in_project` of subtask is set to `True` once the subtask is archived
+        """
+        project = self.env['project.project'].create({
+            'name': 'Project',
+        })
+        task = self.env['project.task'].create([
+            {
+                'name': 'Task',
+                'project_id': project.id,
+                'display_in_project': True,
+            },
+        ])
```

Comment:
no need to create it in with the batch method. since we have to create the task one by one, this is more appropriate.
```py
 task = self.env['project.task'].create({
                'name': 'Task',
                'project_id': project.id,
                'display_in_project': True,
})
```
Also, are you sure you need to create a new project and a new task for that ? Isn't the 'project_pigs' and the task_1 good enough ? Those are common data define here https://github.com/odoo/odoo/blob/master/addons/project/tests/test_project_base.py . When we can, its better to just those default data. It makes the test shorter and easier to maintain. Like when a generic change is made, we can simply update the data in the testprojectcommon instead of every test individually (granted this case is pretty rare, but still)

Reply:
You are totally right, it's better to use data already defined when possible :) I made the adjustments!